### DISCUSSION TITLE
VCF validation using HTSJDK

### DISCUSSION THREAD

**Discussion by makenzi-nzau** [samtools/htsjdk#1705] [General]
Hi!
During development of my application, I encountered a broken VCF where the reserved `DP` was overloaded as

`##FORMAT=<ID=DP,Number=1,Type=Float,Description="Average coverage of the bins">`

and had correspondly rows with `DP`-values such as
`...GT:GQ:CN:DP	1/1:200:0:374.517`.

This causes a crash due to 
`java.lang.NumberFormatException: For input string: "374.517"`.
which is a valid outcome, since the VCF standard specifies that the reserved `DP`-field be an integer. 

However, this type of a message is problematic as a feedback to an end-user.
My question is, which features of HTSJDK 3.0.3 are designed for VCF header validation and specific row value validation?
Or is there recommended way of 'reprocessing' the exception above to carry a more helpful message?

Thanks in advance,
Makenzi.

**Comment by lindenb** [comment]
it's a Locale problem.
try 

`LC_ALL=en_US.UTF-8 java ...`
